packages/wrangler/src/api/startDevWorker/RemoteRuntimeController.ts

```diff
@@ -1,30 +1,303 @@
+import assert from "assert";
+import chalk from "chalk";
+import { Mutex } from "miniflare";
+import {
+	createPreviewSession,
+	createWorkerPreview,
+} from "../../dev/create-worker-preview";
+import {
+	createRemoteWorkerInit,
+	getWorkerAccountAndContext,
+	handleUserFriendlyError,
+} from "../../dev/remote";
+import { UserError } from "../../errors";
+import { logger } from "../../logger";
+import { getAccessToken } from "../../user/access";
 import { RuntimeController } from "./BaseController";
+import { castErrorCause } from "./events";
 import { notImplemented } from "./NotImplementedError";
+import {
+	convertBindingsToCfWorkerInitBindings,
+	MissingConfigError,
+	unwrapHook,
+} from "./utils";
+import type {
+	CfPreviewSession,
+	CfPreviewToken,
+} from "../../dev/create-worker-preview";
+import type { ParseError } from "../../parse";
 import type {
 	BundleCompleteEvent,
 	BundleStartEvent,
 	PreviewTokenExpiredEvent,
 	ReloadCompleteEvent,
 	ReloadStartEvent,
 } from "./events";
+import type { Trigger } from "./types";
 
 export class RemoteRuntimeController extends RuntimeController {
+	#abortController = new AbortController();
```

Comment:
I copied the pattern of use AbortController for cancellations from miniflare (/wrangler?) but, to be useful, we need to check `abortController.aborted` after each `await`

Reply:
We can wrap that in a `this.checkAborted()` call and have that as a one-liner after each await

Reply:
I modified the flow to use abort controllers in the same way as the existing flow—happy to change to a different approach, but I'm not quite sure what you mean here? Why do we need to check `abortController.aborted`?

Reply:
After each await (each async step) the abort controller could've become aborted. Unless we check `abortController.aborted` and return/throw if it is true, the code will continue to run even though it doesn't need to (it's an optimisation but the entire usage of AbortController is essentially an optimisation)

Reply:
I don't think that's needed here. The primary async operations are `fetch` calls, and we pass in the abort signal to those (and handle the thrown abort errors)